central/graphql/resolvers/node_vulnerabilities_test.go

```diff
@@ -0,0 +1,600 @@
+	s.NoError(err)
+	s.Equal("1.1", fixedBy)
+
+	scopedCtx = scoped.Context(ctx, scoped.Scope{
+		Level: v1.SearchCategory_NODE_COMPONENTS,
+		ID:    "comp2#1.1#",
+	})
+	vuln = s.getNodeVulnerabilityResolver(scopedCtx, "cve-2018-1#")
+
+	fixedBy, err = vuln.FixedByVersion(ctx)
+	s.NoError(err)
+	s.Equal("1.5", fixedBy)
+
+	scopedCtx = scoped.Context(ctx, scoped.Scope{
+		Level: v1.SearchCategory_NODE_COMPONENTS,
+		ID:    "comp2#1.1#",
+	})
+	vuln = s.getNodeVulnerabilityResolver(scopedCtx, "cve-2017-1#")
+
+	fixedBy, err = vuln.FixedByVersion(ctx)
+	s.NoError(err)
+	s.Equal("", fixedBy)
+}
+
+func (s *GraphQLNodeVulnerabilityTestSuite) TestNodeVulnerabilitiesScoped() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	node := s.getNodeResolver(ctx, "id1")
+	expected := int32(3)
+
+	vulns, err := node.NodeVulnerabilities(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(expected, int32(len(vulns)))
+	idList := getIDList(ctx, vulns)
+	s.ElementsMatch(idList, []string{"cve-2018-1#", "cve-2019-1#", "cve-2019-2#"})
+
+	count, err := node.NodeVulnerabilityCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(expected, count)
+
+	counter, err := node.NodeVulnerabilityCounter(ctx, RawQuery{})
+	s.NoError(err)
+	checkVulnerabilityCounter(s.T(), counter, expected, 1, 1, 0, 1, 1)
+
+	node = s.getNodeResolver(ctx, "id2")
+	expected = int32(5)
+
+	vulns, err = node.NodeVulnerabilities(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(expected, int32(len(vulns)))
+	idList = getIDList(ctx, vulns)
+	s.ElementsMatch(idList, []string{"cve-2018-1#", "cve-2019-1#", "cve-2019-2#", "cve-2017-1#", "cve-2017-2#"})
+
+	count, err = node.NodeVulnerabilityCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(expected, count)
+
+	counter, err = node.NodeVulnerabilityCounter(ctx, RawQuery{})
+	s.NoError(err)
+	checkVulnerabilityCounter(s.T(), counter, expected, 1, 1, 2, 1, 1)
+}
+
+func (s *GraphQLNodeVulnerabilityTestSuite) TestNodeVulnerabilityMiss() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	vulnID := graphql.ID("invalid")
+
+	_, err := s.resolver.NodeVulnerability(ctx, IDQuery{ID: &vulnID})
+	s.Error(err)
+}
+
+func (s *GraphQLNodeVulnerabilityTestSuite) TestNodeVulnerabilityHit() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	vulnID := graphql.ID("cve-2018-1#")
+
+	vuln, err := s.resolver.NodeVulnerability(ctx, IDQuery{ID: &vulnID})
+	s.NoError(err)
+	s.Equal(vulnID, vuln.Id(ctx))
+}
+
+func (s *GraphQLNodeVulnerabilityTestSuite) TestTopNodeVulnerabilityUnscoped() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	_, err := s.resolver.TopNodeVulnerability(ctx, RawQuery{})
+	s.Error(err)
+}
+
+func (s *GraphQLNodeVulnerabilityTestSuite) TestTopNodeVulnerability() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	node := s.getNodeResolver(ctx, "id2")
+
+	_, err := node.TopNodeVulnerability(ctx, RawQuery{})
+	s.NoError(err)
+
+	// TODO CVSS not populated in node_cves table. Figure out how to test this
```

Comment:
Because the test nodes are upserted directly into stores and not via datastores.